qiskit_nature/problems/second_quantization/lattice/lattice/hyper_cubic_lattice.py

```diff
@@ -0,0 +1,325 @@
+
+
+def _bulk_edges(
+    size: Tuple[int, ...], edge_parameter: Tuple[complex, ...]
+) -> List[Tuple[int, int, complex]]:
+    """Return a list consisting of the edges in th bulk, which don't cross the boundaries.
+
+    Args:
+        size : Lengths of each dimension.
+        edge_parameter : Weights on the edges in each direction.
+
+    Returns:
+        List[Tuple[int, int, complex]] : List of weighted edges that don't cross the boundaries.
+    """
+    list_of_edges = []
+    dim = len(size)
+    coordinates = list(product(*map(range, size)))
+    # add edges excluding the boundary edges
+    for coord in np.array(coordinates):
+        for i in range(dim):
+            if coord[i] != size[i] - 1:
+                relative_vector = np.eye(dim, dtype=int)[i]
+                node_a = _coordinate_to_index(coord, size)
+                node_b = _coordinate_to_index(coord + relative_vector, size)
+                list_of_edges.append((node_a, node_b, edge_parameter[i]))
+    return list_of_edges
+
+
+def _boundary_edges(
+    size: Tuple[int, ...],
+    edge_parameter: Tuple[complex, ...],
+    boundary_condition: Tuple[BoundaryCondition, ...],
+) -> List[Tuple[int, int, complex]]:
+    """Return a list consisting of the edges that cross the boundaries
+        depending on the boundary conditions.
+
+    Args:
+        size : Lengths of each dimension.
+        edge_parameter : Weights on the edges in each direction.
+        boundary_condition : Boundary condition for each dimension.
+            The available boundary conditions are: BoundaryCondition.OPEN and BoundaryCondition.PERIODIC.
+    Raises:
+        ValueError: Given boundary condition is invalid values.
+    Returns:
+        List[Tuple[int, int, complex]]: List of weighted edges that cross the boundaries.
+    """
+    list_of_edges = []
+    dim = len(size)
+    for i in range(dim):
+        # add edges when the boundary condition is periodic.
+        # when the boundary condition in the i-th direction is periodic,
+        # it makes sense only when size[i] is greater than 2.
+        if boundary_condition[i] == BoundaryCondition.PERIODIC:
+            if size[i] <= 2:
+                continue
+            size_list = list(size)
+            size_list[i] = 1
+            coordinates = list(product(*map(range, size_list)))
+            relative_vector = np.eye(dim, dtype=int)[i]
+            for coord in np.array(coordinates):
+                node_b = _coordinate_to_index(coord, size)
+                node_a = _coordinate_to_index((coord - relative_vector) % size, size)
+                list_of_edges.append((node_b, node_a, edge_parameter[i].conjugate()))
+
+        elif boundary_condition[i] != BoundaryCondition.OPEN:
+            raise ValueError(
+                f"Invalid `boundary condition` {boundary_condition[i]} is given."
+                "`boundary condition` must be `BoundaryCondition.OPEN` or `BoundaryCondition.PERIODIC`."
```

Comment:
Thanks for the pointing it out. To avoid hard-coded strings here, I've thought I can write the line119 as
``"`boundary condition` must be " + " or ".join(str(bc) for bc in BoundaryCondition)``, 
which will give rise to 
`` "`boundary condition` must be BoundaryCondition.OPEN or BoundaryCondition.PERIODIC"``.
How about it ?

Reply:
I have modified the message of the ValueError as I suggested above, and the same for `TriangularLattice`.